test/test_mcmc.py

```diff
@@ -0,0 +1,64 @@
+from numpy.testing import assert_allclose
+
+import jax.numpy as np
+import jax.random as random
+from jax import jit, lax
+from jax.scipy.special import expit
+
+import numpyro.distributions as dist
+from numpyro.distributions.util import validation_disabled
+from numpyro.mcmc import hmc_kernel
+from numpyro.util import scan
+
+
+def test_unnormalized_normal():
+    true_mean, true_std = 1., 2.
+    warmup_steps, num_samples = 1000, 8000
+
+    def potential_fn(z):
+        return 0.5 * np.sum(((z[0] - true_mean) / true_std) ** 2)
+
+    def kinetic_fn(r, m_inv):
+        return 0.5 * np.sum(m_inv * r[0] ** 2)
+
+    init_kernel, sample_kernel = hmc_kernel(potential_fn, kinetic_fn)
+    init_samples = [np.array([0.])]
+    hmc_state = init_kernel(init_samples,
+                            num_warmup_steps=warmup_steps)
+    sample_kernel = jit(sample_kernel)
+    hmc_states = lax.scan(sample_kernel, hmc_state, np.arange(num_samples))
+    zs = hmc_states.z[0]
+    assert_allclose(np.mean(zs), true_mean, rtol=0.05)
+    assert_allclose(np.std(zs), true_std, rtol=0.05)
+
+
+def test_logistic_regression():
+    N, dim = 3000, 3
+    warmup_steps, num_samples = 1000, 8000
+    data = random.normal(random.PRNGKey(0), (N, dim))
+    true_coefs = np.arange(1., dim + 1.)
+    probs = expit(np.sum(true_coefs * data, axis=-1))
+    labels = dist.bernoulli(probs).rvs()
```

Comment:
nit: we need `random_state` here

Reply:
@fehiepsi - We are using the scipy sampler here (for all discrete distributions currently), so we don't need to pass in `random_state`, but let me do that for consistency.

Reply:
I see. I mean that we'd better set `random_state=0` here. Right now, `dist.bernoulli(0.5).rvs()` returns different values for different runs with the same numpy.seed:
```
onp.random.seed(0)
dist.bernoulli(0.5).rvs(size=(100,))
```

Reply:
That's really strange that even with a fixed numpy seed it returns different results.

Reply:
Yes, I think it is strange too. While debugging, I thought that `grad` or `value_and_grad` gives inconsistency results but it turns out that the data changed. ^^!